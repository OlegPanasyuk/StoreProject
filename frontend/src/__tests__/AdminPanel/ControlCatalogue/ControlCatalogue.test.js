import React from 'react';
import
ControlCatalogueConnected,
{ ControlCatalogue }
    from '../../../AdminPanel/ControlCatalogue/ControlCatalogue';
import { shallow, configure, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import PropTypes from 'prop-types';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

configure({ adapter: new Adapter() });

function setupComponent(props = {
    addCatalogueItemClose: () => { },
    editCatalogueItem: () => { }
}) {
    let component = shallow(<ControlCatalogue {...props} />);
    return { component, props };
}

describe('ControlCatalogue --- Snapshot', () => {
    it('Component must be rendered', () => {
        let { component } = setupComponent();
        expect(ControlCatalogue.propTypes.editCatalogueItem).toBe(PropTypes.func);
        expect(shallowToJson(component)).toMatchSnapshot();
    });
});