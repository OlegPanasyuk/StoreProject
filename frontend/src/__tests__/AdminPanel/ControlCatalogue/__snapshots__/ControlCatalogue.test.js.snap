// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlCatalogue --- Snapshot Component must be rendered 1`] = `
<Container
  as="div"
  fluid={false}
>
  <Bootstrap(Row)>
    <Col
      as="div"
    >
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        type="button"
        variant="primary"
      >
        Add
      </Button>
    </Col>
  </Bootstrap(Row)>
  <Bootstrap(Row)>
    <Col
      as="div"
      xs="3"
    >
      <ConnectFunction
        data={Array []}
        editCatalogueItem={[Function]}
        name="data"
        showEditForm={[Function]}
      />
    </Col>
    <Col
      as="div"
    />
  </Bootstrap(Row)>
</Container>
`;
